packages/frontend/src/components/markdown-editor/markdown-editor.tsx

```diff
@@ -87,24 +88,31 @@ export const MarkdownEditor = ({ contents, onContentsChange, uploadFile }: Props
   useEffect(() => {
     async function handlePasteEvent(arg: any) {
       const items = (arg.clipboardData || arg.originalEvent.clipboardData).items;
+
+      // Handle all image types on paste event
       if (items[0].type.startsWith('image/')) {
-        const blob = items[0].getAsFile();
-        if (uploadFile) {
-          const name = blob.name.startsWith('image.')
-            ? blob.name.replace('image', `pasted-${nanoid().substring(0, 6)}`)
-            : blob.name.replaceAll(' ', '-');
-          const cursorPos = aceEditorRef.current.editor.getCursorPosition();
-          aceEditorRef.current.editor.session.insert(cursorPos, `![${name}](${name})`);
-
-          const data = await uploadFile(blob, name);
-          if (data) {
-            console.log(`Image pasted with name: '${data.uploadSingleFile.name}'`);
-          }
+        const file = items[0].getAsFile();
+
+        // Set the proper image name by generating a random name if it has a default name (eg: 'image.png')
+        // or add `<` and `>` characters to the name in case it contains any spaces
+        const name = file.name.startsWith('image.')
+          ? file.name.replace('image', `pasted-${nanoid().substring(0, 6)}`)
+          : file.name.contains(' ')
+          ? `<${file.name}>`
+          : file.name;
+
+        // Insert the image with Markdown notation based on the current cursor position
+        const cursorPos = aceEditorRef.current.editor.getCursorPosition();
+        aceEditorRef.current.editor.session.insert(cursorPos, `![${name}](${name})`);
+
+        const data = uploadFile ? await uploadFile(file, name) : null;
+        if (data) {
+          console.log(`Image pasted with name: '${data.uploadSingleFile.name}'`);
         }
```

Comment:
I don't think we need this anymore, since there's an on-screen error if it doesn't work.